presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp

```diff
@@ -219,6 +255,12 @@ std::optional<TypedExprPtr> tryConvertCast(
     return args[0];
   }
 
+  if (returnType.find(kVarchar) == 0 &&
```

Comment:
Please add a comment that only varchar type is truncated. Non-varchar types are not truncated.

Also do you handle a case where we are going from varchar(m) to varchar(n) ?

Reply:
Comment is added.

Yes, this condition includes the case of casting from varchar(m) to varchar(n), by `args[0]->type()->kind() == TypeKind::VARCHAR`. varchar(m) is also parsing as ScalarType<TypeKind::VARCHAR>, as below if I'm looking at the right place. https://github.com/facebookincubator/velox/blob/037c2befd7bce504c0a09ccc777d578206e057a5/velox/type/fbhive/HiveTypeParser.cpp#L108-L120

Reply:
I see.. Seems like the HiveTypeParser translates VARCHAR(n) to VARCHAR. It will need to be enhanced to handle VARCHAR(n) in the future.

Reply:
I see.. Seems like the HiveTypeParser translates VARCHAR(n) to VARCHAR. It will need to be enhanced to handle VARCHAR(n) in the future.

Reply:
I see.. Seems like the HiveTypeParser translates VARCHAR(n) to VARCHAR. It will need to be enhanced to handle VARCHAR(n) in the future.